liquigraph-core/src/main/java/org/liquigraph/core/io/GraphJdbcConnector.java

```diff
@@ -10,12 +10,27 @@
 
 public class GraphJdbcConnector implements LiquigraphJdbcConnector {
 
+    private final LockGuardian lockGuardian;
+
+    public GraphJdbcConnector(LockGuardian lockGuardian) {
+        this.lockGuardian = lockGuardian;
```

Comment:
Is the explicit injection really needed?

My guess is that you've done it for testing purposes, but then I'd just expose a no-argument constructor for regular users:
```java
public GraphJdbcConnector() {
    this(new LockGuardian());
}

@VisibleForTesting
GraphJdbcConnector(LockGuardian lockGuardian) {
    this.lockGuardian = lockGuardian;
}
```

Unless you expect regular users (as in classes, not humans) will want to disable the lock in some scenarios?